modules/core/src/main/java/org/locationtech/jts/index/strtree/STRtree.java

```diff
@@ -263,7 +263,30 @@ protected Comparator getComparator() {
     BoundablePair bp = new BoundablePair(this.getRoot(), this.getRoot(), itemDist);
     return nearestNeighbour(bp);
   }
-  
+  /**
+   * Finds k items in this tree which are the top k nearest neighbors to the given {@link Object}, 
+   * using {@link ItemDistance} as the distance metric.
+   * A Branch-and-Bound tree traversal algorithm is used
+   * to provide an efficient search. This method implements the KNN algorithm described in the following paper:
+   * Roussopoulos, Nick, Stephen Kelley, and Frédéric Vincent. "Nearest neighbor queries." ACM sigmod record. Vol. 24. No. 2. ACM, 1995.
+   * <p>
+   * The query <tt>object</tt> does <b>not</b> have to be 
```

Comment:
Well, the query object actually is a Java Object class although it is used as "Item" in ItemBoundable. I am following the same name convention of the original NearestNeighbor() function.